Makefile

```diff
@@ -0,0 +1,31 @@
+.PHONY: help
+help: ## Show this help
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.venv/bin/python:
+	@uv venv .venv
+
+.venv/.install.stamp: .venv/bin/python pyproject.toml
+	@uv sync
+	@touch .venv/.install.stamp
+
+.PHONY: clean
+clean: ## Remove dependencies, tool caches, and build artifacts
+	@rm -rf .pytest_cache .ruff_cache dist .venv .mypy_cache
+
+.PHONY: format
+format: .venv/.install.stamp ## Run code formatting tools (update in place)
+	@.venv/bin/ruff format . && .venv/bin/ruff check --fix .
+
+
+.PHONY: init
+init: .venv/.install.stamp ## Set up virtual environment and install dependencies
+
+.PHONY: lint
+lint: .venv/.install.stamp ## Run code linting tools
+	@.venv/bin/ruff check . && .venv/bin/ruff format --check
+
+.PHONY: test
+test: .venv/.install.stamp ## Run server tests
+	@.venv/bin/pytest .
```

Comment:
I'm using tox with uv which allows me to test all the Python versions locally.

https://github.com/willkg/everett/blob/main/pyproject.toml#L101-L143

Also, I don't explicitly reference the venv that uv creates because users can set uv up to create the venv in other places. You can use `uv run` to run things in the venv.

https://github.com/willkg/everett/blob/main/justfile#L6-L12

However, we do want to have a `.gitignore` that ignores the `.venv` directory if it's there.